classes/ZipAction.php

```diff
@@ -165,14 +183,39 @@ public function extract(string $from_file, string $to_dir): bool
         }
 
         for ($i = 0; $i < $zip->numFiles; ++$i) {
-            if (!$zip->extractTo($to_dir, [$zip->getNameIndex($i)])) {
+            $nameIndex = $zip->getNameIndex($i);
+            $zip->getExternalAttributesName($nameIndex, $opsys, $stat);
+
+            /*
+             * Symlink case. Matches octal value 12000 and does not end with a separator
+             * @see https://discuss.python.org/t/how-info-zip-represents-symlinks/4104/2
+             * @see https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT - 4.5.7 -UNIX Extra Field (0x000d):
+             */
+            if ($opsys === ZipArchive::OPSYS_UNIX && $this->isCompressedFileASymLink($stat, $nameIndex)) {
+                try {
+                    $this->filesystem->symlink($zip->getFromIndex($i), $to_dir . DIRECTORY_SEPARATOR . $nameIndex);
+                } catch (IOException $e) {
+                    $this->logger->error(
+                        $this->translator->trans(
+                            'Could not extract symbolic link %file% from archive: %error%',
+                            [
+                                '%file%' => $nameIndex,
+                                '%error%' => $e->getMessage(),
+                            ]
+                        )
+                    );
+                    $zip->close();
+
+                    return false;
+                }
```

Comment:
we raise an error log on the Task side if the operation fails, so we could instead use a warning log here
```suggestion
                    $this->logger->warning(
                        $this->translator->trans(
                            'Could not extract symbolic link %file% from archive: %error%',
                            [
                                '%file%' => $nameIndex,
                                '%error%' => $e->getMessage(),
                            ]
                        )
                    );
                    $zip->close();

                    return false;
                }
```

Reply:
I will update this according to the changes of https://github.com/PrestaShop/autoupgrade/pull/1333